### ISSUE TITLE
gem server, ruby stable

### ISSUE THREAD

**Issue by MSP-Greg** [rubygems/rubygems#1864]
I would like to suggest a feature.

`gem server` appears to be broken in 2.6.8, which is included in Ruby 2.4.0.  See [larskanis/rubyinstaller2 issue 21](https://github.com/larskanis/rubyinstaller2/issues/21).

`gem server` generates the 'home page' fine, but links into individual gem docs are incorrect.  Updating to 2.6.10 solved the issue for two testers.

I don't know how everything works with backports to Ruby, could/will 2.6.10 be added to the next Ruby release?

**Comment by bronzdoc** [comment]
Hi @MSP-Greg probably will be shipped, not %100 sure since we(rubygems) don't control that.

**Comment by MSP-Greg** [comment]
@bronzdoc 

Thank you.  I wasn't sure if you 'pushed' or they 'pulled'.  Two questions --

1. Are the tests for rubygems the same in both ruby/ruby and rubygems/rubygems?  If not, is one considered more thorough?

2. I'm a windows type, and working with MinGW / MSYS2 builds.  I have a few failures related to rubygems.  Shown below.  I need to do more testing, I've just worked with ruby trunk, and need get working builds of 2.4, 2.3 and maybe 2.2 stable branches to check for backports.  So the question is - related to test failures from ruby/ruby's tests, should I open issues here or in ruby/ruby?

```
  6) Failure:
TestGemRequestSetLockfile#test_to_s_gem_dependency_non_default [D:/GitHub/ruby/test/rubygems/test_gem_request_set_lockfile.rb:239]:
--- expected
+++ actual
@@ -22,5 +22,5 @@
 " +
 "  a
 " +
-"  b
+"  b (>= 1)
 "


  7) Failure:
TestGemResolver#test_raises_dependency_error [D:/GitHub/ruby/test/rubygems/test_gem_resolver.rb:425]:
--- expected
+++ actual
@@ -1,2 +1,2 @@
-[Gem::Dependency.new("c", Gem::Requirement.new(["= 2"]), :runtime),
- Gem::Dependency.new("c", Gem::Requirement.new(["= 1"]), :runtime)]
+[Gem::Dependency.new("c", Gem::Requirement.new(["= 1"]), :runtime),
+ Gem::Dependency.new("c", Gem::Requirement.new(["= 2"]), :runtime)]


  8) Failure:
TestGemResolver#test_raises_when_possibles_are_exhausted [D:/GitHub/ruby/test/rubygems/test_gem_resolver.rb:524]:
Expected: "c-2"
  Actual: "c-1"
```

**Comment by bronzdoc** [comment]
1. At one point they are, but rubygems/rubygems has the latest rubygems code
2. Rubygems test failures should be reported here, but maybe you want to run the test suite directly from rubygems/rubygems master branch to see if the failures you are getting in ruby/ruby are already fixed.

**Comment by MSP-Greg** [comment]
@bronzdoc 

As I mentioned, I need to do more testing/investigating.  Since the tests pass in ruby/ruby (travis-ci), the problem is probably specific to windows / MinGW builds...  I don't want to submit any issues until I've tested outside of `test-all` and checked against all current  'stable' versions.

**Comment by hsbt** [comment]
I created backport request to ruby trunk tracker.

see https://bugs.ruby-lang.org/issues/13308

**Comment by hsbt** [comment]
backport request was accepted. https://svn.ruby-lang.org/cgi-bin/viewvc.cgi?view=revision&revision=57952

This issue is fixed at Ruby 2.4.1(Maybe, March or April release.)

**Comment by MSP-Greg** [comment]
Thank you for your effort..
